From 2df9c08864eade75f49a3daaf0f74a70335446ea Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Tue, 28 May 2024 10:00:00 +0200
Subject: [PATCH] arm64: dts: st: align DK resmem configuration

Align DK resmem configuration with the beta EV1 configuration.

Change-Id: Icaa01631d6a24d0d3b6f07f039a22ce7fa672d88
Signed-off-by: Alexandre Torgue <alexandre.torgue@foss.st.com>
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/383053
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../st/stm32mp257f-dk-ca35tdcid-resmem.dtsi   | 49 ++++++++-----------
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts     |  2 +-
 2 files changed, 21 insertions(+), 30 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk-ca35tdcid-resmem.dtsi b/arch/arm64/boot/dts/st/stm32mp257f-dk-ca35tdcid-resmem.dtsi
index 889d7067154e..a23386a1a8d0 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk-ca35tdcid-resmem.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk-ca35tdcid-resmem.dtsi
@@ -1,4 +1,4 @@
-// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
 /*
  * Copyright (C) STMicroelectronics 2022 - All Rights Reserved
  * Author: Loic Pallardy loic.pallardy@foss.st.com for STMicroelectronics.
@@ -7,7 +7,7 @@
 /*
  * STM32MP25 reserved memory device tree configuration
  * Project : open
- * Generated by XLmx tool version 2.2 - 7/4/2023 9:06:24 AM
+ * Generated by XLmx tool version 2.2 - 3/6/2024 11:20:05 AM
  */
 
 / {
@@ -17,7 +17,7 @@ reserved-memory {
 		ranges;
 
 		/* Internal RAM reserved memory declaration */
-		sysram1: sysram1@a000000 {
+		tfa_bl31: tfa-bl31@a000000 {
 			reg = <0x0 0xa000000 0x0 0x20000>;
 			no-map;
 		};
@@ -42,28 +42,23 @@ bsec_mirror: bsec-mirror@a040000 {
 			no-map;
 		};
 
-		mcuram1: mcuram1@a041000 {
+		cm33_sram1: cm33-sram1@a041000 {
 			reg = <0x0 0xa041000 0x0 0x1f000>;
 			no-map;
 		};
 
-		mcuram2: mcuram2@a060000 {
-			reg = <0x0 0xa060000 0x0 0x1e000>;
+		cm33_sram2: cm33-sram2@a060000 {
+			reg = <0x0 0xa060000 0x0 0x20000>;
 			no-map;
 		};
 
-		tfa_bl2_heap: tfa-bl2-heap@a07e000 {
-			reg = <0x0 0xa07e000 0x0 0x2000>;
+		cm33_retram: cm33-retram@a080000 {
+			reg = <0x0 0xa080000 0x0 0x1f000>;
 			no-map;
 		};
 
-		retram: retram@a080000 {
-			reg = <0x0 0xa080000 0x0 0x20000>;
-			no-map;
-		};
-
-		vderam: vderam@a0a0000 {
-			reg = <0x0 0xa0a0000 0x0 0x20000>;
+		ddr_param: ddr-param@a09f000 {
+			reg = <0x0 0xa09f000 0x0 0x1000>;
 			no-map;
 		};
 
@@ -74,7 +69,7 @@ pcie_device: pcie-device@10000000 {
 		};
 
 		/* Backup RAM reserved memory declaration */
-		bkpspram1: bkpspram1@42000000 {
+		bl31_lowpower: bl31-lowpower@42000000 {
 			reg = <0x0 0x42000000 0x0 0x1000>;
 			no-map;
 		};
@@ -92,25 +87,21 @@ mm_ospi1: mm-ospi@60000000 {
 
 		/* DDR reserved memory declaration */
 		tfm_code: tfm-code@80000000 {
-			compatible = "shared-dma-pool";
 			reg = <0x0 0x80000000 0x0 0x100000>;
 			no-map;
 		};
 
 		cm33_cube_fw: cm33-cube-fw@80100000 {
-			compatible = "shared-dma-pool";
 			reg = <0x0 0x80100000 0x0 0x800000>;
 			no-map;
 		};
 
 		tfm_data: tfm-data@80900000 {
-			compatible = "shared-dma-pool";
 			reg = <0x0 0x80900000 0x0 0x100000>;
 			no-map;
 		};
 
 		cm33_cube_data: cm33-cube-data@80a00000 {
-			compatible = "shared-dma-pool";
 			reg = <0x0 0x80a00000 0x0 0x800000>;
 			no-map;
 		};
@@ -140,12 +131,12 @@ vdev0buffer: vdev0buffer@812fa000 {
 		};
 
 		spare1: spare1@81300000 {
-			reg = <0x0 0x81300000 0x0 0xcf0000>;
+			reg = <0x0 0x81300000 0x0 0xcc0000>;
 			no-map;
 		};
 
-		bl31_context: bl31-context@81ff0000 {
-			reg = <0x0 0x81ff0000 0x0 0x10000>;
+		bl31_context: bl31-context@81fc0000 {
+			reg = <0x0 0x81fc0000 0x0 0x40000>;
 			no-map;
 		};
 
@@ -154,18 +145,18 @@ op_tee: op-tee@82000000 {
 			no-map;
 		};
 
-		gpu_reserved: gpu-reserved@fb000000 {
-			reg = <0x0 0xfb000000 0x0 0x4000000>;
+		gpu_reserved: gpu-reserved@fa800000 {
+			reg = <0x0 0xfa800000 0x0 0x4000000>;
 			no-map;
 		};
 
-		ltdc_sec_layer: ltdc-sec-layer@ff000000 {
-			reg = <0x0 0xff000000 0x0 0x800000>;
+		ltdc_sec_layer: ltdc-sec-layer@fe800000 {
+			reg = <0x0 0xfe800000 0x0 0x800000>;
 			no-map;
 		};
 
-		ltdc_sec_rotation: ltdc-sec-rotation@ff800000 {
-			reg = <0x0 0xff800000 0x0 0x800000>;
+		ltdc_sec_rotation: ltdc-sec-rotation@ff000000 {
+			reg = <0x0 0xff000000 0x0 0x1000000>;
 			no-map;
 		};
 
diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
index b05d55890dcd..92841cfc0f20 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -508,7 +508,7 @@ &m33_rproc {
 	memory-region = <&cm33_cube_fw>, <&cm33_cube_data>,
 			<&ipc_shmem_1>, <&vdev0vring0>,
 			<&vdev0vring1>, <&vdev0buffer>,
-			<&mcuram2>;
+			<&cm33_sram2>;
 	st,syscfg-nsvtor = <&a35ss_syscfg 0xa8 0xffffff80>;
 	status = "okay";
 };
-- 
2.39.5

